source/acquaman/AMContinuousScanActionController.cpp

```diff
@@ -0,0 +1,513 @@
+		AMDetector *oneDetector = AMBeamline::bl()->exposedDetectorByInfo(generalConfig_->detectorConfigurations().at(x));
+		asScalerChannelDetector = qobject_cast<CLSScalerChannelDetector*>(oneDetector);
+		if(asScalerChannelDetector){
+			scalerChannelIndexMap.insert(asScalerChannelDetector->enabledChannelIndex(), oneDetector->name());
+			scalerChannelVectors.insert(oneDetector->name(), QVector<qint32>(rebasedTotalCount, 0));
+			scalerChannelRunningSums.insert(oneDetector->name(), 0);
+		}
+	}
+
+	// I think we can do a check against the number of actually enabled channels on the scaler ... just can't figure out a good way to get that value right now
+//	AMDSLightWeightScalarDataHolder *asScalarDataHolder = qobject_cast<AMDSLightWeightScalarDataHolder*>(scalerClientDataRequst->data().at(0));
+//	if(asScalarDataHolder && asScalarDataHolder->dataArray().constVectorQint32().count() != ){
+//		qDebug() << "FATAL ERROR, MISMATCH BETWEEN RECEIVED DATA SIZE OF SCALER AND REQUESTED SIZE. RECEIVED: " << asScalarDataHolder->dataArray().constVectorQint32().count() << "REQUESTED: " << scalerChannelVectors.count();
+//		return;
+//	}
+
+	for(int x = 0; x < totalCount; x++){
+		AMDSLightWeightScalarDataHolder *asScalarDataHolder = qobject_cast<AMDSLightWeightScalarDataHolder*>(scalerClientDataRequst->data().at(x));
+		if(asScalarDataHolder){
+			QVector<qint32> oneVector = asScalarDataHolder->dataArray().constVectorQint32();
+
+			int tempRunningSum;
+			QString channelString;
+			for(int y = 0, ySize = oneVector.count(); y < ySize; y++){
+				channelString = scalerChannelIndexMap.value(y);
+				tempRunningSum = scalerChannelRunningSums.value(channelString);
+				tempRunningSum += oneVector.at(y);
+				scalerChannelRunningSums[channelString] = tempRunningSum;
+
+				if( (((x+1)*baseScalerTimeScale) % largestBaseTimeScale) == 0){
+					int rebaseIndex = (x*baseScalerTimeScale)/largestBaseTimeScale;
+					(scalerChannelVectors[channelString])[rebaseIndex] = scalerChannelRunningSums.value(channelString);
+					scalerChannelRunningSums[channelString] = 0;
+				}
+			}
+		}
+	}
+
+	AMDSClientDataRequest *oneAmptekDataRequest = clientDataRequestMap_.value("Amptek SDD 240");
+
+	int initiateMovementIndex = 0;
+	bool foundMovementStart = false;
+	bool foundMovementEnd = false;
+
+	// Loop backwards from the end to find the start of the movement we're interested in
+	for(int x = oneAmptekDataRequest->data().count()-1; (x > 0) && !foundMovementStart; x--){
+		AMDSDwellSpectralDataHolder *dataHolderAsDwellSpectral = qobject_cast<AMDSDwellSpectralDataHolder*>(oneAmptekDataRequest->data().at(x));
+		int encoderPulsesInPeriod = dataHolderAsDwellSpectral->dwellStatusData().generalPurposeCounter();
+		if(!foundMovementEnd && encoderPulsesInPeriod > 20){
+			qDebug() << "Found movement end at index " << x;
+			foundMovementEnd = true;
+		}
+		else if(foundMovementEnd && encoderPulsesInPeriod < 1){
+			foundMovementStart = true;
+			initiateMovementIndex = x;
```

Comment:
The data request could return more data than just the move we're interested in. This should return the index of the start of the move assuming that the "last move" in the buffer is the one we're interested in. I believe that assumption has to be valid in the paradigm we're using.